### PR TITLE
WIP PLAT-145913

### PR DESCRIPTION
**Pull request by skanjila**
- Initial set of unit tests stubbed out
- Partly filled in schema unit tests, fill continue to flush these out

### CODE REVIEW

File: aepp/datasets.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def __init__(
             aepp.config.endpoints["global"] + aepp.config.endpoints["dataset"]
         )
 
-    def getLabels(self, dataSetId: str = None) -> dict:
+    def getLabelSchemaTests(self, dataSetId: str = None) -> dict:
```
Comment (pitchmuc) [inline aepp/datasets.py:91]: Why the change  in name here ? Would it not make sense to keep the `getLabels` method ?

**Comment by kratika2210** [comment]
lgtm!

File: aepp/datasets.py (hunk 1 above)
Comment (cmenguy) [inline aepp/datasets.py:91]: This doesn't seem correct, why are we renaming this function?

File: tests/schema_test.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+from aepp.schema import Schema
+import unittest
+from unittest.mock import patch, MagicMock
+
+
+class SchemaTest(unittest.TestCase):
```
Comment (cmenguy) [inline tests/schema_test.py:6]: It would be useful to go even further in these tests, where we mock not just the function call, but the actual functionality - for exaple for `test_schema_get_tenant_id` can we mock the actual return value from the API and use that to validate that the tenant ID value we get is indeed the expected one?

For example see https://realpython.com/testing-third-party-apis-with-mocks/#mocking-the-complete-service-behavior for how to do this.

**Review by cmenguy — CHANGES_REQUESTED** [review]
It's a good start but I think it would be good to mock not just the function call but also the actual data return to validate functionally - see my comment for more details. Also it seems you may have inadvertently renamed a function that should not have been renamed.

**Comment by cmenguy** [comment]
Also can you update the `README.md` for how one can run these unit tests from the command line?

File: tests/schema_test.py (hunk 2 above)
Comment (skanjila) [reply tests/schema_test.py:6]: yes thanks for the great feedback, will dig into next level of detail

File: aepp/datasets.py (hunk 1 above)
Comment (skanjila) [reply aepp/datasets.py:91]: ya this was an accident, will fix, sorry about that

Comment (skanjila) [reply aepp/datasets.py:91]: Ok this should be fixed now, can you please recheck this

File: tests/schema_test.py (hunk 2 above)
Comment (pitchmuc) [reply tests/schema_test.py:6]: It would be great to test that as I, and I assume you as well, will probably build business logic out of the default response from the AEP API.

File: aepp/datasets.py (hunk 1 above)
Comment (pitchmuc) [reply aepp/datasets.py:91]: I did not see that change, you probably did not commit to a new PR. 
I am fixing it, I will merge this PR for now, as a start and will fix the typo myself via another commit

**Review by pitchmuc — APPROVED** [review]
Accepted this PR as base for unit test. I will fix the typo we noticed. 
We will build on top of that setup.
@skanjila Can you also document the running of the test and if there is any process to know in order to participate on creating new tests. Personally, I am not familiar with it so it would be awesome.
